### PR TITLE
[#515, #524] nodrop, clone, and dirty-checking fixes

### PR DESCRIPTION
**Pull request by ppuleo**
- Use angular.copy when cloning nodes
- Simplify nodrop, clone, and isDirty checking in $uiTreeHelper dragInfo
- Update $uiTreeHelper dragInfo.apply to allow null parent.$nodeScope for single-level trees

### CODE REVIEW

**Comment by ppuleo** [comment]
Hmm, looks like the tests are starting in the middle of the build. I think [runSequence](https://www.npmjs.com/package/run-sequence) might be a better than gulp.start for the build task.

**Comment by Voles** [comment]
Thanks for this, @ppuleo! CircleCI isn't well configured so don't mind the failing build. As long as Travis succeeds.

Can you elaborate your improvements, please? Do you need these for a project you're working on?

**Comment by ppuleo** [comment]
Thanks for reviewing @Voles! And thanks for maintaining this project. This PR addresses two issues:

1. Use angular.copy when cloning a node to a different tree. I described the issue in #524, and added an [example](http://embed.plnkr.co/8vLEYEb4nyTuE3LpNiPP/preview). 

2. Simplify the data-nodrop-enabled and drop-in-place cases in the `$uiTreeHelper` service `dragInfo.apply` method. The logic in there was getting hard to follow and there was already a method defined to check for a node dropped in its original place (`isDirty`). Based on the conversation in #515, this PR enforces cancellation of the drop if `data-nodrop-enabled` is added to either the destination ui-tree or the immediate ui-tree-nodes parent element.

I'm using angular-ui-tree in a project now so I'll try to contribute fixes for any bugs I find. So far, it generally works well.

**Comment by Voles** [comment]
@ppuleo thanks for your detailed explanation! I've tested your changes, LGTM!
Because of the v2.7.0 release, there are conflicts with your branch.

If you resolve these conflicts I'll merge your changes! Sorry for the inconvenience...

**Comment by ppuleo** [comment]
@Voles Sure, no problem. I should probably add example as well. I'll update and comment again when this is ready to review.

**Comment by Voles** [comment]
@ppuleo great! Feel free to add the examples as a separate PR. Maybe I can help you with this as well.

**Comment by Voles** [comment]
I'd like to suggest to update the current connected-trees example page. By adding some information about cloning the node and using input-fields for the node name, things should be clear enough. WDYT?

**Comment by ppuleo** [comment]
@Voles ok, I've added examples for 'data-nodrop-enabled` and cloning between connected trees so I think this is ready for review. Thanks!

**Comment by vitorenesduarte** [comment]
+1

**Comment by Voles** [comment]
Merged. Thanks very much, @ppuleo! The examples are very useful for other developers!
